### PR TITLE
EditEngagement Endpoint

### PR DESCRIPTION
**Pull request by McGarvs**

### CODE REVIEW

**Comment by McGarvs** [comment]
Somehow the login branch got in here, remove before pullint

File: cradlepoint/pages/api/editEngagement.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  try{
+    const data = req.body;
+    const valid = await engagementSchema.isValid(data);
+    if (valid && ObjectId.isValid(data.testPlanId)){
+      const result = engagementSchema.cast(data);
+      // Set ID strings to Mongo ObjectId's
+      const newId = ObjectId(result._id);
+      const newTestPlanId = ObjectId(result.testPlanId);
+      const query = {_id: newId};
```
Comment (mtran-12) [inline cradlepoint/pages/api/editEngagement.js:20]: can we name these id and testPlanId? the new is misleading

File: cradlepoint/schemas/engagementSchema.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
 import * as yup from 'yup';
+const { ObjectId } = require('bson');
```
Comment (mtran-12) [inline cradlepoint/schemas/engagementSchema.js:2]: I don't think that this is actually used anywhere

File: cradlepoint/pages/api/getActiveEngagements.js [hunk 3]
```diff
@@ -6,9 +6,8 @@ export default async (req, res) => {
   // All status codes greater than 13 are for inactive engagements
   const query = { statusCode: { $lt: 13 } };
 
-  const client = await connectToDb();
-  const cursor = await client.collection("engagements").find(query);
-  
+  const db = await connectToDb();
+  const cursor = db.collection("engagements").find(query);
```
Comment (mtran-12) [inline cradlepoint/pages/api/getActiveEngagements.js:10]: Please don't change multiple things in one PR. Each PR should only handle one issue at a time

**Review by mtran-12 — COMMENTED** [review]
Please don't change multiple things in one PR. It makes it more difficult to verify that everything works and to actually review the code.

**Comment by McGarvs** [comment]
{
  "_id": "617249b199915be1b771aca1", 
  "statusCode": 2,
  "engagementDetails": "Test Details 2",
  "SE": "1234",
  "POC_Engineer": "2345",
  "customer": "Small Finance",
  "SFDC": "https://cradlepoint.lightning.force.com/lightning/r/Opportunity/006380...",
  "BOM": [""],
  "testPlanId": "61724e5599915be1b771acb2",
  "createdOn": "2021-11-20T15:00:00.000Z"
}

**Pull request merged**